### PR TITLE
[ENH] Add support for plotting intervals in `plot_series()`

### PR DESCRIPTION
**Pull request by chillerobscuro**
#### Reference Issues/PRs

This would close [Issue-3824](https://github.com/sktime/sktime/issues/3824) as well as [Issue-541](https://github.com/sktime/sktime/issues/541). There have been multiple other PRs for the original issue but as far as I can see they're all closed ([545](https://github.com/sktime/sktime/pull/545), [802](https://github.com/sktime/sktime/pull/802)).

#### What does this implement/fix? Explain your changes.

Allows user to pass `pred_interval` to `plot_series()` and plot the confidence interval provided in that argument.

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

Any other edge cases I need to check for? So far there is pretty rigorous check of the schema of the interval DataFrame, and there's a check to ensure the index matches that of the final series passed to the function.

Once there's some buy in for this approach I'll add at least one test case.

I'm not sure what killed the other attempts at this feature, it's been discussed for at least two years. I think we start simple and then expand to the `predict_quantiles()`, multivariate, plotting kwargs, and other more complex cases after.

### CODE REVIEW

**Comment by chillerobscuro** [comment]
Almost all of the pieces were there but with this PR I think `sktime` has the most impressive <10 lines of code in open source:
```
from sktime.datasets import load_airline
from sktime.forecasting.arima import AutoARIMA
from sktime.utils import plotting

y = load_airline()
forecaster = AutoARIMA(sp=12)

forecaster.fit(y, fh=list(range(1, 13)))
y_pred = forecaster.predict()
pred_int = forecaster.predict_interval(coverage=.95)
plotting.plot_series(y, y_pred, labels=["y", 'y_pred'], pred_interval=pred_int)
```
image

**Comment by fkiraly** [comment]
> Almost all of the pieces were there but with this PR I think sktime has the most impressive <10 lines of code in open source:

You're too kind 😊 

Let's see whether we can improve on that with the pipelines tutorial at pydata global 2022 :-)

**Comment by fkiraly** [comment]
> And IMO this (with image) should be the marquee code block for the docs, either on the home or "get started" page.

Hm, you're probably right - maybe with a pipeline creation step or so?

File: sktime/utils/validation/forecasting.py
Comment (fkiraly) [inline sktime/utils/validation/forecasting.py:460]: some of this is redundant with

```python
from sktime.datatypes import check_is_mtype

check_is_mtype(interval_df, "pred_interval")
```

(I think all except the "index to match" check)

**Review by fkiraly — DISMISSED** [review]
Makes sense as an improvement.

Regarding the checkers, I left a non-blocking comment.

In addition, I just looked at the checker at the top - it seems to call `check_y` which enforces univariate inputs, and may also be replaceable by `check_is_scitype`, but I would not consider the checks part of this PR's scope.

**Comment by fkiraly** [comment]
> I'm not sure what killed the other attempts at this feature, it's been discussed for at least two years. 

Maybe just bad luck?

I can imagine that it was also the lack of a clean design for prediction interval return objects that has been introduced in the beginning of 2022 and was then presented at pydata Berlin 2022 (and/or good documentation around the probabilistic prediction interface which was also added around that time).

**Comment by chillerobscuro** [comment]
> > And IMO this (with image) should be the marquee code block for the docs, either on the home or "get started" page.
> 
> Hm, you're probably right - maybe with a pipeline creation step or so?

I think since any sequence of steps is conceptually a "pipeline" that could be foregone for the "marquee" and expanded upon later, for the sake of conciseness.

File: sktime/utils/validation/forecasting.py
Comment (chillerobscuro) [reply sktime/utils/validation/forecasting.py:460]: I will check that out and replace these lines, good tip

Comment (chillerobscuro) [reply sktime/utils/validation/forecasting.py:460]: After looking into it, I am leaning towards leaving the checks as is. I would typically want to reuse `check` style code but the existing code raises very informative errors and the function needs to exist regardless, at least to check the column name (used in the plotting function) and the matching index as you mentioned. Is that acceptable?

Comment (fkiraly) [reply sktime/utils/validation/forecasting.py:460]: sure, but that would mean you don´t find `check_is_mtype` informative of the same degree, which should be improved then, separately.

Comment (chillerobscuro) [reply sktime/utils/validation/forecasting.py:460]: I was not fully aware of the usefulness of `return_metadata=True`, now that I see it I've added it in. I kept checks for index matching, exact column name as used in the plot function ("Coverage"), and that there is only one pair of interval series. Should be good to go if this passes tests.

FYI you mentioned a "non blocking" comment but it appears this PR can't be completed until all comments are resolved. It's very possible I misunderstood what you meant though.

Comment (fkiraly) [reply sktime/utils/validation/forecasting.py:460]: in the end, a core developer needs to merge it either way

**Review by fkiraly — APPROVED** [review]

**Pull request merged**
